internal/mailer/accessor.go

```diff
@@ -36,18 +36,28 @@ func (p *postgresEmailStatusAccessor) WriteEmailStatus(_ context.Context, es Ema
 func (p *postgresEmailStatusAccessor) GetAll(ctx context.Context, spec GetAllEmailStatusSpec) (*AccessorGetAllPaginationData, error) {
 	paginationArgs := database.BuildPaginationArgs(spec.PaginationSpec)
 
-	// Initialize clauses and arguments
 	var (
+		whereClauses    []string
 		joinClauses     []string
 		extraClauses    []string
 		args            []interface{}
+		countArgs       []interface{}
 		argsIndex       = 1
 		extraClausesRaw = []string{
 			"LIMIT $%d",
 			"OFFSET $%d",
 		}
 	)
 
+	// Build WHERE clauses
```

Comment:
Why you put the logic in accessor instead of service?

Reply:
If we do it in the service instead, the pagination will not work

Reply:
Could you elaborate on why the pagination wouldn’t work if the logic is placed in the service?

Reply:
Pagination logic is placed in the accessor layer because it directly involves database queries with LIMIT and OFFSET clauses, which are most efficiently handled by the database itself. This ensures only the required data is fetched, reducing memory usage and improving performance. Keeping pagination in the accessor aligns with the principle of separation of concerns, where the accessor is responsible for data retrieval and the service layer focuses on business logic. Additionally, placing the logic in the accessor makes it reusable across multiple services, avoiding duplication.

If the pagination logic were moved to the service layer, the accessor would fetch all records from the database, and the service would then trim the data. This approach is inefficient as it wastes database and memory resources by loading unnecessary data. Moreover, databases are optimized to handle paginated queries efficiently, and bypassing this capability leads to slower performance, especially with large datasets. For these reasons, pagination is best implemented in the accessor.